client/galaxy/scripts/mvc/workflow/workflow-manager.js

```diff
@@ -7,10 +7,49 @@ define(['mvc/workflow/workflow-connector'], function( Connector ) {
         this.name = null;
         this.has_changes = false;
         this.active_form_has_changes = false;
+        this.nodeLabels = {}; // TODO: track and enforce output labels also
     }
     $.extend( Workflow.prototype, {
-        create_node: function ( type, title_text, tool_id ) {
-            var node = this.app.prebuildNode( type, title_text, tool_id );
+        canLabelNodeWith: function( label ) {
+            if( label ) {
+                return ! (label in this.nodeLabels);
```

Comment:
I would use the following, which is very easy to read for me. The choice is yours.
` return label in this.nodeLabels ? false : true;`